source/fundamentals/indexes.txt

```diff
@@ -0,0 +1,253 @@
+
+
+See the MongoDB server manual section on :manual:`Compound indexes </core/index-compound>` for more information.
+
+Multikey Indexes (Indexes on Array Fields)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Multikey indexes** are indexes that improve performance for queries that specify ascending or descending indexes on
+fields that contain an array value. You can define a multikey index using the same syntax as a single field or compound
+index.
+
+The following example use the ``createIndex()`` method to create a compound, multikey index on the ``rated``,
+``genres`` (an array of Strings), and ``title`` field in the ``movies`` collection in the ``sample_mflix`` database.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin multikey index
+   :end-before: end multikey index
+
+The following is an example of a query that would be covered by the index created above.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin covered multikey query
+   :end-before: end covered multikey query
+
+Multikey indexes behave differently from non-multikey indexes in terms of query coverage, index bound computation, and
+sort behavior. For a full explanation of multikey indexes, including a discussion of their behavior and limitations,
+refer to the :manual:`Multikey Indexes page </core/index-multikey>` in the MongoDB manual.
+
+Text Indexes
+~~~~~~~~~~~~
+
+**Text indexes** support text search queries on string content. These indexes can include any field whose value is a
+string or an array of string elements. MongoDB supports text search for various languages. You can specify the default
+language as an option when creating the index.
+
+The following example uses the ``createIndex()`` method to create a ``text`` index on the ``fullplot`` field in the
+``movies`` collection in the ``sample_mflix`` database.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin text index
+   :end-before: end text index
+
+The following is an example of a query that would be covered by the index created above. Note that the ``sort`` is
+omitted because text indexes do not contain sort order.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/IndexPage.java
+   :language: java
+   :dedent:
+   :start-after: begin text query
+   :end-before: end text query
+
+For a full explanation of text search with MongoDB, refer to :manual:`Text Indexes </core/index-text>` in the MongoDB
+manual.
+
+Geospatial Indexes
+~~~~~~~~~~~~~~~~~~
+
+MongoDB supports queries of geospatial coordinate data using **2dsphere indexes**. With a 2dsphere index, you can query
+the geospatial data for inclusion, intersection, and proximity. For more information on querying geospatial data, see
+:manual:`Geospatial Queries </geospatial-queries/>`.
```

Comment:
This and other sections include the "For a full explanation" sentence at the very end of the section that already links to this page. The Node driver documentation contains a page on geospatial queries: https://docs.mongodb.com/drivers/node/fundamentals/crud/read-operations/geo
Would it make more sense to link to the Java driver docs equivalent page here, since this link is rendered redundant by the "for a full explanation" link at the end of the section?